### ISSUE TITLE
Gulp inject not inserting correctly in mustache template

### ISSUE THREAD

**Issue by nebulousGirl** [klei/gulp-inject#136]
When I run `gulp inject-css`, the first everything is ok. But, when I run the command again the results is weird, the inject tag gets emptied out and code is randomly duplicated at the end of the file.

**head.mustache:**
```mustache
<!DOCTYPE html>
<html class="no-js" lang="{{locale}}">
    <head>
        <meta charset="utf-8">
        <title>{{sitename}} | {{meta.title}}</title>
        <meta name="viewport" content="width=device-width, initial-scale=1">

            <!-- bower:css -->
            <!-- endinject -->

        {{#stylesheets_external}}
        <link rel="stylesheet" type="text/css" href="{{{src}}}" media="{{media}}" />
        {{/stylesheets_external}}
        {{#stylesheets}}
        <link rel="stylesheet" type="text/css" href="{{{assets_url}}}{{{src}}}" media="{{media}}" />
        {{/stylesheets}}
        {{#scripts_head}}
        <script type="text/javascript" src="{{{assets_url}}}{{{.}}}"></script>
        {{/scripts_head}}
    </head>
    <body class="PAGE">
        <!--[if lte IE 8]>
            <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</p>
        <![endif]--> 
```

**gulpfile.js:**
```js
var gulp = require("gulp"),
    inject = require('gulp-inject'),
    bower_files = require('bower-files')();

var css_inject = function (filepath, file, i, length) {
    return '<link rel="stylesheet" href="' + filepath.replace('/bower_components', '{{{ assets_url }}}/vendor') + '" />';
};
var js_inject = function (filepath, file, i, length) {
    return '<script type="text/javascript" src="' + filepath.replace('/bower_components', '{{{ assets_url }}}/vendor') + '"></script>';
};
gulp.task('inject-css', function() {
        return gulp.src('./views/**/*.mustache').pipe(inject(gulp.src(bower_files.ext('css').files, {read: false, base: './bower_components' }), {name: 'bower', transform: css_inject}))
                .pipe(gulp.dest('./views'));
});
gulp.task('inject-js', function() {
        return gulp.src('./views/**/*.mustache').pipe(inject(gulp.src(bower_files.ext('js').files, {read: false, base: './bower_components' }), {name: 'bower', transform: js_inject}))
                .pipe(gulp.dest('./views'));
});

```

**Comment by nebulousGirl** [comment]
Rolled back to 1.5.0 after others having issues with injecting and it works correctly.

**Comment by joakimbeng** [comment]
Duplicate of #135 which is solved in `3.0.0`
